test/integration/domain/timeout/index.test.js

```diff
@@ -0,0 +1,69 @@
+const ProxyCache = require('../../../../src/lib/proxyCache')
+const { PAYER_DFSP, PARTY_ID_TYPE, PROXY_NAME } = require('../../constants')
+const fixtures = require('../../../fixtures')
+const { ProxyApiClient } = require('../../../util')
+
+const wait = (ms) => new Promise(resolve => setTimeout(resolve, ms))
+
+describe('Timeout Handler', () => {
+  let proxyCache
+  const proxyClient = new ProxyApiClient()
+
+  beforeAll(async () => {
+    proxyCache = await ProxyCache.getConnectedCache()
+    const redisClient = proxyCache.redisClient
+    const redisNodes = redisClient.nodes ? redisClient.nodes('master') : [redisClient]
+    await Promise.all(redisNodes.map(async node => {
+      await node.flushall()
+    }))
+  })
+
+  afterAll(async () => {
+    return Promise.all([
+      proxyClient.deleteHistory(),
+      proxyCache.disconnect()
```

Comment:
One of disadvantages of using _ProxyCache_ singleton is that we can't now run integration tests in parallel, coz one test might finish and disconnect cache, while another test is trying to read or write some data (and will fail)

Reply:
Refactored.

I don't think we can run integration tests in parallel if the datastore and mock-server history is shared among tests.